functorch/experimental/_cond.py

```diff
@@ -125,11 +125,34 @@ def cond_dense(pred, true_fn, false_fn, operands):
 def cond_autograd(pred, true_fn, false_fn, *operands):
     # TODO: support autograd
     flat_operands, _ = tree_flatten([true_fn, false_fn] + [operands])
-    assert all(not f.requires_grad for f in flat_operands
-               if isinstance(f, torch.Tensor))
+
+    requires_grad = any(
+        isinstance(arg, torch.Tensor) and arg.requires_grad
+        for arg in flat_operands
+    )
 
     with _ExcludeDispatchKeyGuard(DispatchKeySet(DispatchKey.AutogradCPU)):
-        return cond(pred, true_fn, false_fn, *operands)
+        result = cond(pred, true_fn, false_fn, *operands)
+
+        # If there is requires_grad, we delay the error until backward pass
+        if requires_grad:
+            # cond can only return one value
+            err_fn = torch._C._functions.DelayedError(
+                b"trying to differentiate buffers inside torch.cond",
+                1,
+            )
```

Comment:
nit: I'd phrase this as "NYI: torch.cond doesn't support autograd"

Reply:
will do